openmetadata-ui/src/main/resources/ui/src/hooks/useTableFilters.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ *  Copyright 2025 Collate.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+import { useMemo } from 'react';
+import { useHistory } from 'react-router-dom';
+import useCustomLocation from './useCustomLocation/useCustomLocation';
+
+type FilterState = Record<string, string | boolean>;
+
+export const useTableFilters = <T extends FilterState>(initialFilters: T) => {
+  const location = useCustomLocation();
+  const history = useHistory();
+
+  const parseFiltersFromUrl = (): T => {
+    const searchParams = new URLSearchParams(location.search);
```

Comment:
Can we make use of this lib? https://github.com/ljharb/qs

Reply:
Updated. Using  useLocationSearch hook, which is using the qs libarary.